pyfarm/agent/utility.py

```diff
@@ -196,7 +196,11 @@ def dumps(*args, **kwargs):
     else:
         obj = dict(*args, **kwargs)
 
-    return _dumps(obj, default=default_json_encoder, indent=indent)
+    dump = _dumps(obj, default=default_json_encoder, indent=indent)
+
+    # Many parts of Twisted will not accept the typical output
+    # from _dumps, which is unicode, so we encode into ascii here.
+    return dump.encode("ascii", "replace")
```

Comment:
That doesn't seem right to me. We should be able to send data in any encoding we want, as long as we present it as a stream of bytes. Limiting ourselves to ascii because of a limitation in Twisted seems seriously wrong to me.